requests_mock/response.py

```diff
@@ -170,6 +171,10 @@ def create_response(request, **kwargs):
 
     response = _http_adapter.build_response(request, raw)
     response.connection = connection
+
+    header_encoding = get_encoding_from_headers(response.headers)
+    if header_encoding:
+        encoding = header_encoding
```

Comment:
Hey, I've had a chance to sit down and look through this. It's not an area of the code that changes frequently any more. 

Do you think it makes more sense to move that discovery up? So the way that content discovery cascades currently only things that you pass through as a byte string or lower end up without an encoding. JSON and text both end up as UTF-8. 

Maybe we should put 
```
   headers = kwargs.pop('headers', {})
   encoding = get_encoding_from_headers(CaseInsensitiveDict(headers))
```
at L148 and like a 
```
content = text.encode(encoding or 'utf-8')
```
at L159? (though that might be considered a change in behaviour, leave as is?)

Two interesting tests to run at the moment: 

```
   def test_json_without_charset_gets_utf8(self):
       data = {'hello': 'world'}
       resp = self.create_response(json=data)

       self.assertEqual(data, resp.json())
       self.assertIsInstance(resp.text, six.string_types)
       self.assertIsInstance(resp.content, six.binary_type)
       self.assertIsInstance(resp.json(), dict)
       self.assertEqual(resp.encoding, 'utf-8')

   def test_json_with_iso_charset(self):
       data = {'hello': 'world'}
       headers = {'content-type': 'application/json;charset=ISO-8859-1'}
       resp = self.create_response(json=data, headers=headers)

       self.assertEqual(data, resp.json())
       self.assertIsInstance(resp.text, six.string_types)
       self.assertIsInstance(resp.content, six.binary_type)
       self.assertIsInstance(resp.json(), dict)
       self.assertEqual(resp.content_type, 'application/json')
       self.assertEqual(resp.encoding, 'ISO-8859-1')
```

the first one currently passes, it really could go either way. The second one fails (existing as well) and should pass.

Thoughts?